flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/utils/SqlGatewayServiceExtension.java

```diff
@@ -43,6 +47,15 @@ public class SqlGatewayServiceExtension implements BeforeAllCallback, AfterAllCa
     private SqlGatewayService service;
     private SessionManager sessionManager;
     private TemporaryFolder temporaryFolder;
+    private final Supplier<Configuration> configSupplier;
```

Comment:
It seems that `Configuration ` is unmodifiable from MiniClusterResource? why here use supplier?

Reply:
Yes. But the MiniCluster is created in the beforeAll that is later than create the `SqlGatewayServiceExtension`.

Reply:
Yes. But the MiniCluster is created in the beforeAll that is later than create the `SqlGatewayServiceExtension`.